src/nvim/testdir/test_filechanged.vim

```diff
@@ -0,0 +1,149 @@
+" Tests for when a file was changed outside of Vim.
+
+func Test_FileChangedShell_reload()
+  if !has('unix')
+    return
+  endif
+  augroup testreload
+    au FileChangedShell Xchanged_r let g:reason = v:fcs_reason | let v:fcs_choice = 'reload'
+  augroup END
+  new Xchanged_r
+  call setline(1, 'reload this')
+  write
+  " Need to wait until the timestamp would change by at least a second.
+  sleep 2
+  silent !echo 'extra line' >>Xchanged_r
```

Comment:
I guess that the changes to `Xchanged_r` are not reflected in the buffer after `:checktime` so the assertions consistently fail.

Reply:
This failure is actually because there is no UI active, so `do_dialog()` just chooses the default option.